src/pyseekdb/client/configuration.py

```
@@ -42,17 +43,28 @@ class HNSWConfiguration:
+    properties: dict[str, str | int | float | bool] | None = None
 
     def __post_init__(self):
         if self.dimension <= 0:
             raise ValueError(f"dimension must be positive, got {self.dimension}")
         valid_distances = [e.value for e in DistanceMetric]
         if self.distance not in valid_distances:
             raise ValueError(f"distance must be one of {valid_distances}, got {self.distance}")
+        if self.properties:
+            for value in self.properties.values():
+                if not isinstance(value, (str, int, float, bool)):
+                    raise TypeError(f"properties must be a dictionary of string, int, float, or bool, got {value}")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)